nav2_bt_navigator/src/bt_navigator.cpp

```diff
@@ -221,6 +226,9 @@ BtNavigator::navigateToPose()
         RCLCPP_INFO(get_logger(), "Received goal preemption request");
         action_server_->accept_pending_goal();
         initializeGoalPose();
+        blackboard_->set<bool>("new_goal_received", true);
+      } else {
+        blackboard_->set<bool>("new_goal_received", false);
```

Comment:
Overall, this param isn't necessary, Aitor's implementation just cached the last path and compared them, we should go that route. I'd suggest removing this BT node from your PR and letting his land first and then use his to add this work for time / distance replanning with new goal.